packages/dd-trace/src/plugins/util/web.js

```diff
@@ -337,6 +351,59 @@ const web = {
 
     return `${path}?${qs}`
   },
+
+  extractIp (context) {
+    const { req, config } = context
+
+    if (config.clientIpHeaderDisabled) return null
+
+    const headers = req.headers
+    let foundIp
+
+    if (config.clientIpHeader) {
+      const header = headers[config.clientIpHeader]
+      if (!header) return null
+
+      foundIp = findFirstIp(header)
+    } else {
+      const foundHeaders = []
+
+      for (let i = 0; i < ipHeaderList.length; i++) {
+        if (headers[ipHeaderList[i]]) {
+          foundHeaders.push(ipHeaderList[i])
+        }
+      }
+
+      if (foundHeaders.length === 0) {
+        foundIp = req.socket && req.socket.remoteAddress
+      } else if (foundHeaders.length === 1) {
+        const header = headers[foundHeaders[0]]
+        const firstIp = findFirstIp(header)
+
+        foundIp = firstIp || (req.socket && req.socket.remoteAddress)
+      } else if (foundHeaders.length > 1) {
+        const tags = {}
+        let headersList = ''
+
+        for (let i = 0; i < foundHeaders.length; i++) {
+          const headerName = foundHeaders[i]
+
+          headersList = headersList ? `${headersList},${headerName}` : headerName
+
+          tags[`${HTTP_REQUEST_HEADERS}.${headerName}`] = headers[headerName]
```

Comment:
Why add this when not explicitly configured since there are the other tags?

Reply:
This is a case when the IP resolution cannot be done because multiple headers are present, so it reports which headers are present in tags so the user can solve the problem.

Reply:
But then why have also a separate tag for the headers list? And also the user might not know that this is a problem and might just assume we're always sending these tags. It's fine if we want to report this and have a proper use case for it, but it shouldn't be a replacement for proper helpful errors and shouldn't be confusing.

Reply:
Since the "duplicate ip header" tag starts with `_dd` I think it's hidden by default, and is meant to be analysed by the backend to show an alert to the user. Once again this is specification level and not implementation level is it ?